### ISSUE TITLE
**3.0.X** dash multiperiod timeline playback failure, **2.5.19** works, dashif **3.2.0** works

### ISSUE THREAD

**Issue by arturpilaszkiewicz** [google/shaka-player#3139]
Hi,
I've got a problem regarding stream playback of dash timeline with multiperiod change every ~60s

shaka **3.0.X** (**3.0.0**, **3.0.1**, **3.0.2**, **3.0.3**, **3.0.4**, **3.0.5**, **3.0.6**, **3.0.7**)
https://shaka-player-demo.appspot.com/demo/#audiolang=en-GB;textlang=en-GB;uilang=en-GB;asset=//r.dcs.dev.redlabs.pl/livedash/o2/ATM-Lab/zos/usr/ap/07.livx;panel=CUSTOM%20CONTENT;build=uncompiled

 - plays a bit of content
 - goes into spinner
 - fragments new fragments are  not acquired, the buffer is emptied
 - plays a bit of content
 - and so on
 - player reports
```
Jumping forward 11.226612210273743 seconds to catch up with the seek range
```
 - playback failure is within ~20s of stream playback
 
shaka **2.5.19**
https://v2-5-19-dot-shaka-player-demo.appspot.com/demo/#audiolang=en-GB;textlang=en-GB;uilang=en-GB;asset=https://r.dcs.dev.redlabs.pl/livedash/o2/ATM-Lab/zos/usr/ap/07.livx;panel=CUSTOM%20CONTENT;build=uncompiled

 - maintains buffer
 - no problems with playback

dashif **3.2.0**
https://reference.dashif.org/dash.js/v3.2.0/samples/dash-if-reference-player/index.html

 - no problem
 
 Could you please help me why this might be happening ?
 I cannot see nothing out of ordinary in changelogs, that could impact playback.
 How should I proceed to debug this playback issue ?

**Comment by TheModMaker** [comment]
Next time please use the bug template for this kind of issue.  It looks like when playing, we cannot find the segment reference we need.  There is some problem with the segment index.

**Comment by apilaszkiewicz** [comment]
Ok thanks, does it mean that we are producing an improper manifest ?

**Comment by TheModMaker** [comment]
Probably not.  I'm referring to our segment index that we create from the manifest.  This is probably a bug on our part.

**Comment by ismena** [comment]
Hi @apilaszkiewicz 
Seems like the content is no longer available. Could you please verify/update the manifest? Thanks!

**Comment by apilaszkiewicz** [comment]
@ismena its back online, we had ~3 hours of downtime due to server error.

**Comment by ismena** [comment]
Look how lucky I am! :)

I found the root cause and am considering the best way to address it. I should have a fix for this next week.

**Comment by ismena** [comment]
Update: the fix is in review. It's failing some tests now, so it might be a few days until it's merged, but it's 100% on track for v3.1 release.
